cli/console.ts

```diff
@@ -129,40 +129,51 @@ export function printCompiledGraph(graph: dataform.ICompiledGraph, asJson: boole
     writeStdOut(successOutput(`Compiled ${actionCount} action(s).`));
     if (graph.tables && graph.tables.length) {
       graph.tables.forEach(setOrValidateTableEnumType);
-      writeStdOut(`${graph.tables.length} dataset(s):`);
-      graph.tables.forEach(compiledTable => {
-        writeStdOut(
-          `${datasetString(
-            compiledTable.target,
-            tableTypeEnumToString(compiledTable.enumType),
-            compiledTable.disabled
-          )}`,
-          1
-        );
-      });
+      writeStdOut(`${graph.tables.length} dataset(s)${quietCompilation ? "." : ":"}`);
+      if(!quietCompilation){
+          graph.tables.forEach(compiledTable => {
+            writeStdOut(
+              `${datasetString(
+                compiledTable.target,
+                tableTypeEnumToString(compiledTable.enumType),
+                compiledTable.disabled
+              )}`,
+              1
+            );
+          });
+      }
     }
     if (graph.assertions && graph.assertions.length) {
-      writeStdOut(`${graph.assertions.length} assertion(s):`);
-      graph.assertions.forEach(assertion => {
-        writeStdOut(assertionString(assertion.target, assertion.disabled), 1);
-      });
+      writeStdOut(`${graph.assertions.length} assertion(s)${quietCompilation ? "." : ":"}`);
+      if(!quietCompilation){
+          graph.assertions.forEach(assertion => {
+            writeStdOut(assertionString(assertion.target, assertion.disabled), 1);
+          });
+      }
     }
     if (graph.operations && graph.operations.length) {
-      writeStdOut(`${graph.operations.length} operation(s):`);
-      graph.operations.forEach(operation => {
-        writeStdOut(operationString(operation.target, operation.disabled), 1);
-      });
+      writeStdOut(`${graph.operations.length} operation(s)${quietCompilation ? "." : ":"}`);
+      if(!quietCompilation){
+          graph.operations.forEach(operation => {
+            writeStdOut(operationString(operation.target, operation.disabled), 1);
+          });
+      }
     }
   }
 }
 
-export function printCompiledGraphErrors(graphErrors: dataform.IGraphErrors) {
+export function printCompiledGraphErrors(graphErrors: dataform.IGraphErrors, quietCompilation: boolean) {
   if (graphErrors.compilationErrors && graphErrors.compilationErrors.length > 0) {
     printError("Compilation errors:", 1);
-    graphErrors.compilationErrors.forEach(compileError => {
+    graphErrors.compilationErrors.forEach((compileError, index) => {
+      let errorMeta = ""
+
+      if (compileError.message.includes("Unexpected identifier")) {
+        errorMeta = compileError.stack.split("\n").slice(0, 3).join("\n");
+      }
```

Comment:
I propose to extract it to a helper private method and call it directly on line 176.

Additionally, we need to add a safety check in the method: make sure that compileError.stack is actually defined and not empty.

Also I propose to left a small comment on what we are doing here.